File: CSharp part1/04. Console-Input-Output/06.QuadraticEquation/QuadraticEquation.cs
/* Write a program that reads the coefficients a, b and c of a quadratic equation ax2 + bx + c = 0 
 * and solves it (prints its real roots). */

using System;

class QuadraticEquation
{
    static void Main()
    {
        Console.WriteLine("Quadratic Equation solver (ax^2+bx+c=0)");
        Console.Write("Enter a:");
        double a = double.Parse(Console.ReadLine());
        if (a == 0)
        {
           Console.WriteLine("\"a\" cannot be zero");
           return;
        }
        Console.Write("Enter b:");
        double b = double.Parse(Console.ReadLine());
        Console.Write("Enter c:");
        double c = double.Parse(Console.ReadLine());
        double D = b * b - 4 * a * c;
        if (D < 0)
        {
            Console.WriteLine("Equation has no real roots");
        }
        else if (D == 0)
        {
            Console.WriteLine("There are two equal roots - x1=x2={0}", (-b) / (2 * a));
        }
        else
        {
           Console.WriteLine("There are two roots - x1={0}; x2={1}", (-b + Math.Sqrt(D))/(2*a), (-b - Math.Sqrt(D))/(2*a));
        }
    }
}

File: JavaScript-Fundamentals/ConditionalStatements/06.QuadraticEquation/QuadraticEquation.js
// Write a script that reads the coefficients a, b and c of a quadratic equation ax2 + bx + c = 0 and solves it (prints its real roots).
// Calculates and prints its real roots.

var jsConsole;
var btn = document.getElementById('solve');
btn.onclick = solveEquation;

function solveEquation () {
	var a = jsConsole.readFloat('#firstNumber');
	var b= jsConsole.readFloat('#secondNumber');
	var c = jsConsole.readFloat('#thirdNumber');
	var D = b * b - 4 * a * c;

	if (a === 0){
		alert(" \"a cannot be zero");
		return;
	}

	if(D < 0){
		jsConsole.writeLine("Equation has no real roots");
	} else if (D === 0) {
		jsConsole.writeLine("There are two equal roots - x1=x2="+ (-b) / (2 * a));
	} else {
		jsConsole.writeLine("There are two roots - x1=" + (-b + Math.sqrt(D))/(2 * a) + "; x2=" + (-b - Math.sqrt(D))/(2*a));
	}
}